lib/rules/_base.js

```diff
@@ -372,18 +389,47 @@ export default class {
     };
   }
 
-  _pushConfig(config) {
-    this._configStack.push(config);
+  _pushConfig(config, disabled, commentNode) {
+    this._configStack.push({ config, disabled, commentNode });
     this.config = config;
+    this.disabledErrors = 0;
   }
 
   _popConfig(config) {
-    if (last(this._configStack) !== config) {
+    if (last(this._configStack).config !== config) {
       throw new Error('Configuration stack out of sync with AST!');
     }
 
+    if (this._configStack.at(-1).disabled && this.disabledErrors === 0) {
```

Comment:
It might make sense to split this into its own method, but I do think `_popConfig` is probably still the place to call it.